Enable default file browsers with different properties (aria label, title...)

HaudinFlorence opened the pull request:
In the filebrowser-extension, this PR aims at modfiying the defaultFileBrowser and in the browserWidget plugins to enable to replace the default filebrowser by a custom one having different properties such as aria label, icon etc...

cc @afshin @DenisaCG 

<!--
Thanks for contributing to JupyterLab!
Please fill out the following items to submit a pull request.
See the contributing guidelines for more information:
https://github.com/jupyterlab/jupyterlab/blob/main/CONTRIBUTING.md
-->

## References

Fixes issue #15689

## Code changes

Move some parts of codes from the widgetBrowser plugin to the defaultFileBrowser one.

## User-facing changes

<!-- Describe any visual or user interaction changes and how they address the issue. -->

<!-- For visual changes, include before and after screenshots or GIF/mp4/other video demo here. -->

## Backwards-incompatible changes

<!-- Describe any backwards-incompatible changes to JupyterLab public APIs. -->

krassowski commented on packages/filebrowser-extension/src/index.ts:
```diff
@@ -312,22 +311,54 @@ const defaultFileBrowser: JupyterFrontEndPlugin<IDefaultFileBrowser> = {
   description: 'Provides the default file browser',
   provides: IDefaultFileBrowser,
-  requires: [IFileBrowserFactory],
+  requires: [IFileBrowserFactory, ITranslator],
```
Compatibility-wise, should translator be optional?

krassowski commented on packages/filebrowser-extension/src/index.ts:
```diff
@@ -313,21 +312,53 @@ const defaultFileBrowser: JupyterFrontEndPlugin<IDefaultFileBrowser> = {
     tree: JupyterFrontEnd.ITreeResolver | null,
-    labShell: ILabShell | null
+    labShell: ILabShell | null,
+    translator: ITranslator
```
```suggestion
    translator: ITranslator | null
```

krassowski commented on packages/filebrowser-extension/src/index.ts:
```diff
@@ -313,21 +312,53 @@ const defaultFileBrowser: JupyterFrontEndPlugin<IDefaultFileBrowser> = {
+    translator: ITranslator
   ): Promise<IDefaultFileBrowser> => {
     const { commands } = app;
+    const trans = translator.load('jupyterlab');
```
```suggestion
    const trans = (translator ?? nullTranslator).load('jupyterlab');
```

`nullTranslator` needs to be imported from `@jupyterlab/translation`

HaudinFlorence replied:
@krassowski Thanks for your suggestions. I indeed forgot to properly treat the possibility of not having a translator.

krassowski reviewed (approved):
Thanks @HaudinFlorence!

> While this is good, the `main` branch still points to 4.1.x so let's wait with merging a little while. I think we usually transition/branch out after first patch release to reduce backporting churn, but let's discuss on the team meeting if we want to handle it differently this time.

HaudinFlorence commented:
please update galata references

krassowski reviewed (changes requested):
@HaudinFlorence I believe the snapshot changes are flaky, can you revert them please?

HaudinFlorence commented:
> @HaudinFlorence I believe the snapshot changes are flaky, can you revert them please?

@krassowski Sorry for the delay. I just reverted the snapshots changes and rebased with the main branch.

HaudinFlorence commented:
@krassowski There are 2 tests failing (Linux Tests/ Linux release_test and the UI Tests / Visual Regression Documentation)

krassowski commented:
My guess is that the visual snapshot is flaky, and the release test just timed out, it happens sometimes:

```
+ node buildutils/lib/local-repository.js start
Starting verdaccio on port 4873 in /tmp/verdaccio
Timed out!
LOG:
```

It would be great if someone took some time to figure out how to increase the timeout slightly.

krassowski reviewed (approved):
Thanks @HaudinFlorence! (tests are all passing after a few restarts)

HaudinFlorence commented:
> Thanks @HaudinFlorence! (tests are all passing after a few restarts)

Thanks Mike @krassowski !
